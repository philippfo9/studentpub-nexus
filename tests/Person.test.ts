import { createTestContext } from './__helpers';

const testCtx = createTestContext();

it('can search for persons', async () => {
    // plain testing

    const personResult = await testCtx.client.send(`
        query {
            searchPersons(searchTerm: "Jeff") { 
                id
                name
            }
        }
    `);

    expect(personResult.searchPersons.length).toBe(1);
});

it('can add a student', async () => {
    const addStudentResult = await testCtx.client.send(`
        mutation {
            addStudent(grade: 8, school: "Alabama County School", personInput: {birthday: 915148800, citizenship: "AT", name: "Kyrven"}) {
                birthday
                citizenship
                id
                name
                school
            }
        }
    `);
    expect(addStudentResult).toMatchSnapshot({
        addStudent: {
            birthday: 915148800,
            citizenship: 'AT',
            id: expect.any(String),
            name: 'Kyrven',
            school: 'Alabama County School'
        }
    })
});