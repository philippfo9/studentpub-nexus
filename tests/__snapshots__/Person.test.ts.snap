// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can add a student 1`] = `
Object {
  "addStudent": Object {
    "birthday": 915148800,
    "citizenship": "AT",
    "id": Any<String>,
    "name": "Kyrven",
    "school": "Alabama County School",
  },
}
`;
